File: demos/javascript/data_structures_demo.js
class Stack {
  #items = [];
  push(value) { this.#items.push(value); }
  pop() { return this.#items.pop(); }
  get size() { return this.#items.length; }
}

const stack = new Stack();
stack.push("parse");
stack.push("render");
const scores = new Map([["Ada", 95], ["Lin", 91]]);
const tags = new Set(["docs", "code", "docs"]);
const bytes = new Uint8Array([72, 105]);
const tree = { value: "root", children: [{ value: "child", children: [] }] };

console.log(`stack=${stack.size}:${stack.pop()}`);
console.log(`scores=${[...scores].map(([k, v]) => `${k}:${v}`).join(",")}`);
console.log(`tags=${[...tags].sort().join(",")}`);
console.log(`typed-array=${bytes.join(",")}`);
console.log(`tree-children=${tree.children.length}`);

File: demos/csharp/DataStructuresDemo.cs
using System;
using System.Collections.Generic;

namespace DataStructuresTopic;

public sealed record Node<T>(T Value, IReadOnlyList<Node<T>> Children);

public static class Program
{
    public static void Main()
    {
        var values = new List<string> { "parse", "validate", "render" };
        var scores = new Dictionary<string, int> { ["Ada"] = 95, ["Lin"] = 91 };
        var tags = new HashSet<string> { "docs", "code", "docs" };
        var tasks = new PriorityQueue<string, int>();
        tasks.Enqueue("docs", 2);
        tasks.Enqueue("tests", 1);
        var tree = new Node<string>("root", new[] { new Node<string>("child", Array.Empty<Node<string>>()) });

        Console.WriteLine($"list={string.Join(',', values)}");
        Console.WriteLine($"scores=Ada:{scores["Ada"]},Lin:{scores["Lin"]}");
        Console.WriteLine($"tags={tags.Count}");
        Console.WriteLine($"next={tasks.Dequeue()}");
        Console.WriteLine($"tree-children={tree.Children.Count}");
    }
}
